roachpb/data.go

```diff
@@ -809,12 +804,18 @@ func (t Transaction) Short() string {
 	return t.ID.String()[:8]
 }
 
+// ResetObservedTimestamps clears out all timestamps recorded from individual
+// nodes.
+func (t *Transaction) ResetObservedTimestamps() {
+	t.ObservedTimestamps = nil
+}
+
 // UpdateObservedTimestamp stores a timestamp off a node's clock for future
 // operations in the transaction. When multiple calls are made for a single
 // nodeID, the lowest timestamp prevails. Calls with timestamps above
-// t.ObservedTimestamp have no effect.
+// t.ObservedTimestamp have no effect, unless t.ObservedTimestamp is zero.
```

Comment:
Btw, I think it's quite confusing to call this variable `ObservedTimestamp`. This is the maximum ceiling for uncertainty interval. It has much less to do with observations. Could we rename this `UncertaintyCeilingTS` or back to `MaxTimestamp`?